src/UpdatedPackagesCollector.js

```diff
@@ -79,19 +79,40 @@ export default class UpdatedPackagesCollector {
   }
 
   isPackageDependentOf(packageName, dependency) {
-    var dependencies = this.packageGraph.get(packageName).dependencies;
+    if (!this.cache[packageName]) {
+      this.cache[packageName] = {};
+    }
+
+    if (this.cache[packageName][dependency] === "dependent") {
+      return true;
+    } else if (this.cache[packageName][dependency] === "visited") {
+      return false;
+    }
+
+    let dependencies = this.packageGraph.get(packageName).dependencies;
 
     if (dependencies.indexOf(dependency) > -1) {
+      this.cache[packageName][dependency] = "dependent";
       return true;
     }
 
-    return !!find(dependencies, dep => {
-      return this.isPackageDependentOf(dep, dependency);
+    this.cache[packageName][dependency] = "visited";
```

Comment:
I think it was the case for if a package A depends on package B, and then package B depends on package A